### PR TITLE
Document user migration API

### PR DESCRIPTION
**Pull request by Pytal**
Close https://github.com/nextcloud/documentation/issues/8025

@jancborchardt @AndyScherzinger should we use "User Migration" or "User migration"?

User_migration_—_Nextcloud_latest_Developer_Manual_latest_documentation_-_2022-04-25_08 46 46

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
I don't remember the name for the casing, but the result is: User migration

**Comment by nickvergessen** [comment]
> should we use "User Migration" or "User migration"?

We only case the first char.

File: developer_manual/digging_deeper/user_migration.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      * Returns the description
+      *
+      * @since 24.0.0
+      */
+    public function getDescription(): string {
+      return $this->l10n->t('My App information');
+    }
+  }
+
+The ``MyAppMigrator`` class needs to be registered during the :ref:`app bootstrap<Bootstrapping>`
```
Comment (Valdnet) [inline developer_manual/digging_deeper/user_migration.rst:125]:
```suggestion
The ``MyAppMigrator`` class needs to be registered during the :ref:`app bootstrap<Bootstrapping>`.
```

**Review by Valdnet — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Pull request merged**